File: apps/api/AssetManagement.Api/Models/AssetHistoryChange.cs
namespace AssetManagement.Api.Models;

public class AssetHistoryChange
{
    public Guid Id { get; set; }

    public Guid AssetHistoryId { get; set; }
    public AssetHistory AssetHistory { get; set; } = null!;

    public string FieldName { get; set; } = string.Empty;
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/AssetHistoryChange.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "asset_history_changes")
class AssetHistoryChange(
    @Id
    @Column(name = "id", columnDefinition = "CHAR(36)")
    var id: UUID = UUID.randomUUID(),

    @Column(name = "asset_history_id", nullable = false, columnDefinition = "CHAR(36)")
    var assetHistoryId: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "asset_history_id", insertable = false, updatable = false)
    var assetHistory: AssetHistory? = null,

    @Column(name = "field_name", nullable = false)
    var fieldName: String = "",

    @Column(name = "old_value", columnDefinition = "TEXT")
    var oldValue: String? = null,

    @Column(name = "new_value", columnDefinition = "TEXT")
    var newValue: String? = null
)
